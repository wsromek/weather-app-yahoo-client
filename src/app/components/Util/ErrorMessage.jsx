import React from 'react';
import PropTypes from 'prop-types';
import { Alert, Button } from 'react-bootstrap';

function ErrorMessage({ error, onErrorReset }) {
  if (!error.message) return null;

  return (
    <Alert bsStyle="danger">
      <p>{error.message}</p>
      <p><Button bsStyle="warning" onClick={onErrorReset}>Close</Button></p>
    </Alert>
  );
}

ErrorMessage.propTypes = {
  error: PropTypes.shape({
    message: PropTypes.string.isRequired,
  }).isRequired,
  onErrorReset: PropTypes.func.isRequired,
};

export default ErrorMessage;
