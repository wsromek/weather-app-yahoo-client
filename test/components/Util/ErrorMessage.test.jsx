import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import ErrorMessage from '../../../src/app/components/Util/ErrorMessage';

describe('ErrorMessage component', () => {
  const error = {
    message: 'test_error',
  };
  const onErrorReset = jest.fn();

  it('renders correctly', () => {
    const tree = renderer.create(
      <ErrorMessage error={error} onErrorReset={onErrorReset} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls onErrorReset when alert is closed', () => {
    const component = mount(<ErrorMessage error={error} onErrorReset={onErrorReset} />);

    component.find('button').simulate('click');

    expect(onErrorReset).toHaveBeenCalled();
  });
});
