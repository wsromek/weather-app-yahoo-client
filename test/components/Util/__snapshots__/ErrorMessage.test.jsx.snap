// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component renders correctly 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  <p>
    test_error
  </p>
  <p>
    <button
      className="btn btn-warning"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </p>
</div>
`;
